packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -1,70 +1,122 @@
+  const [interactions, setInteractions] = useState<ChatInteraction[]>([])
+  const [currentInteraction, setCurrentInteraction] =
+    useState<ChatInteraction>()
+  const [currentResponse, setCurrentResponse] = useState('')
+  // Keep track of the text that the user is currently typing into the textfield
+  const [userChatTextfieldValue, setUserChatTextfieldValue] = useState('')
+  const [initialMessageProcessed, setInitialMessageProcessed] = useState(false)
+  const { mutate: sendChatMessageToAgent } = useSendChatMessageToAgent({
+    onSuccess: data => {
+      // whenever the response is returned, set the last interaction response text
+      setCurrentResponse(data.responseText)
     },
   })
-  const theme = useTheme()
-  const [messages, setMessages] = useState([initialMessage])
-  const [currentMessage, setCurrentMessage] = useState('')
+  // Restore chat session history, if exists.
+  // TODO: currently only a single page is restored.  Add support for multiple pages (and detect the user scrolling up to restore the next page of results older)
+  const { data: sessionHistory, mutate: getSessionHistory } =
+    useGetAgentChatSessionHistory()
+
+  useEffect(() => {
+    // when we have both a message and response, add a new interaction and reset
+    // Note : We want the current interaction to be visible to the user while this async job processes (show the user message, and a loading response).
+    // You might think we could do this work in the onSuccess of createAgentChatInteraction, but I think there's a race condition (currentInteraction may not be set when onSuccess is called).
+    if (currentResponse && currentInteraction) {
+      currentInteraction.chatResponseText = currentResponse
+      setInteractions([...interactions, currentInteraction])
+      setCurrentResponse('')
+      setCurrentInteraction(undefined)
+    }
+  }, [currentResponse, currentInteraction])
 
   useEffect(() => {
     // on mount, create a new agent session!
-    if (createAgentSession) {
+    if (createAgentSession && !agentSession) {
       createAgentSession({
         agentAccessLevel: AgentAccessLevel.PUBLICLY_ACCESSIBLE,
         agentId: agentId,
       })
     }
-  }, [createAgentSession])
+  }, [createAgentSession, agentSession])
+
+  useEffect(() => {
+    // on mount, resolve the initial message chat interaction (if set)
+    if (agentSession && initialMessage && !initialMessageProcessed) {
+      setCurrentInteraction({ userMessage: initialMessage })
+      sendChatMessageToAgent({
+        chatText: initialMessage,
+        sessionId: agentSession!.sessionId,
+      })
+      setInitialMessageProcessed(true)
+    }
+  }, [agentSession, initialMessage, initialMessageProcessed])
+
+  useEffect(() => {
+    // on mount, read a page worth of previous messages
+    if (agentSession && getSessionHistory) {
+      getSessionHistory(agentSession.sessionId)
+    }
+  }, [agentSession, getSessionHistory])
```

Comment:
would expect this to use `useQuery`

If you decide to try the optimistic update approach I suggested earlier, you may need to give it infinite `staleTime`